src/common/io.py

```diff
@@ -1607,9 +1617,10 @@ def get_variables_data(self,
             stop_index = len(time) + start_index
 
         for name in names:
-            trajectories.append(self._get_variable_data_as_trajectory(name, time, start_index, stop_index))
+            trajectories[name] = self._get_variable_data_as_trajectory(name, time, start_index, stop_index)
 
-        new_start_index = start_index + len(time) if len(trajectories) > 0 else None
+        new_start_index = stop_index if len(trajectories) > 0 else None
```

Comment:
The earlier formulation was also wrong, we should just use `stop_index` since we set it accordingly above.

Reply:
Why? What if you do not have enough points saved? I.e. there is only stop_index-3 result points in the result file? Also why you "len(traj)"? That is very confusing...

Reply:
@chria this has been updated again, please see the latest version. I also changed it from `if len(trajectories) > 0` to `if trajectories`, since it will evaluate to `False` if `trajectories = {}`.